app/services/search_service.rb

```diff
@@ -54,13 +54,23 @@ def find_comments(input, limit = 5)
   end
 
   def find_nodes(input, limit = 5)
-    Node.limit(limit)
-      .order('nid DESC')
-      .where('node.status = 1 AND title LIKE ?', '%' + input + '%')
+    if ActiveRecord::Base.connection.adapter_name == 'Mysql2'
+      Node.search(input)
+        .group(:nid)
+        .includes(:node)
+        .references(:node)
+        .limit(limit)
+        .where("node.type": ["note", "page"], "node.status": 1)
+        .order('node.changed DESC').map{ |x| x.parent}
```

Comment:
OK, this is pretty good. But the problem is that then the `.map()` function will run once for each result -- not very efficient! Instead we could do:

```ruby
nids = Node.search ...
```

up on line 58, and then at the end:


```ruby
        .order('node.changed DESC')
        .collect(&:nid)
Node.find nids
```

That way we get them all in just one single query. It's not super pretty, but it's efficient!